### ISSUE TITLE
Beatmap Editor Rework

### ISSUE THREAD

**Issue by Pennek** [ppy/osu-wiki#3128]
As part of #2615 i am splitting the Editor rework out into its own issue as it is a heavy topic with lots of articles to be rewritten/changed.

Tasks:

- [ ] AiMod - overall well-written but text is dated and has some fluff. needs update with mention of flaws. mentions of alternative tools like Modding Assistant would also be helpful.
- [ ] Beat Snap Divisor - overall well-written but could use some formatting changes to cut the length down. needs mention of unsupported beat snap divisors. could possibly merge with [this](https://osu.ppy.sh/community/forums/topics/935026) user created guide
- [ ] Compose - needs a rewrite. images as well. mode-specific sections for this article would be great as its a huge mess right now. potential users reading these are here for one mode anyways. This would be a perfect place to implement https://github.com/ppy/osu-wiki/pull/3097#issuecomment-579857343_
- [ ] Design - i know little about storyboarding. mention of other tools is one thing that should be mentioned here at least
- [ ] Distance Snap - should this even be its own article? mapping community has moved on from using this as the denominator for emphasis. could implement into an "editor tools"-article
- [ ] Kiai Time - overall well-written. needs a bit of cleanup and linking to ranking criteria for some parts. could be merged into the 'Timing' article
- [ ] Menu - maybe not worth having since this is more like an 'editor shortcuts' article, which is a duplicate of everything in wiki/Shortcut_key_reference
- [ ] SB Load - again, i know little about storyboarding. is this relevant?
- [ ] Song Setup - needs update to fit the Ranking criteria. a lot of this needs to reference the metadata ranking criteria section. rewriting this to be a simple explanation of the song setup screen, instead of what is allowed and what isnt might be the way to go
- [ ] Timelines - I can see nothing wrong here. looks good
- [ ] Timing - could use some cleanup but otherwise good

**Comment by Coppertine** [comment]
>  SB Load - again, i know little about storyboarding. is this relevant?
No, it isn't SB Load is to the GPU load, but most performance in Storyboard is just cpu.

**Comment by cl8n** [comment]
even for "gpu load" it's a pretty irrelevant metric these days. it definitely doesn't need an entire article for an explanation

**Comment by Pennek** [comment]
ahh, so that's what pulled me down in my bn app, lol!

**Comment by Pennek** [comment]
what do you suggest we do with that article?

**Comment by cl8n** [comment]
both sound good to me

**Comment by Coppertine** [comment]
> Design - i know little about storyboarding. mention of other tools is one thing that should be mentioned here at least

Needs the Overlay layer thing (the layer that is over the top of hitobjects and key display). Also for programs, Storybrew (developed by an osu!developer, but not associated) is a C# program in which you create using C#, has a preview feature of the storyboard before you export and such,  There is also osupy, SGL and others

**Comment by Coppertine** [comment]
We also need the "Sample import" window (that almost noone except mania and some hitsounders use) It bassically allows you to add in storyboard sounds into the map. The only way to preview them is holding tab in Mania mode of the editor

**Comment by hypercyte** [comment]
This hasn't been updated in a while, but I can work on updating the AiMod page if no one else is currently working on it
